### PR TITLE
Download more than 30 files

### PR DESCRIPTION
**Pull request by MVrachev**
Related to: https://github.com/vmware/precaution/issues/134

In order to download all of the diff files from a pull request, we have to access the data from GitHub page by page.

Also, there is an option to customize how many results do you want per page.

I added a unit test handling the paging but I  had to change the other unit tests a little bit in order to achieve that.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -19,7 +20,11 @@ const sampleOnlyDeletions = require('./fixtures/pull_request.deletions.json')
 const sampleDelAddModif = require('./fixtures/pull_request.deletions.modif.add.json')
 
 function mockPRContents (github, PR) {
-  github.pullRequests.listFiles = jest.fn().mockResolvedValue(PR)
+  github.pullRequests.listFiles = jest.fn(() => {
+    --github.numPagesOfContent
+    return PR
```
Comment (ericwb) [inline test/index.test.js]: We shouldn't use variables in all caps.  It effectively looks like a const

File: test/index.test.js [hunk 2]
```diff
@@ -241,6 +259,25 @@ describe('Bandit-linter', () => {
...
     })
 
+    test('handles paging through the GitHub API', async () => {
+      github.numPagesOfContent = 2
+      await app.receive(pullRequestOpenedEvent)
+
+      expect(github.hasNextPage).toHaveBeenCalledTimes(2)
+      expect(github.getNextPage).toHaveBeenCalledTimes(1)
+
+      const expectedRes = { 'data':
```
Comment (ericwb) [inline test/index.test.js]: put 'data': on the next line

File: github_api_helper.js [hunk 3]
```diff
@@ -26,21 +26,37 @@ function inProgressAPIresponse (context, headSha) {
   })
 }
 
+// Filters the files which are not relevant to us
```
Comment (ericwb) [inline github_api_helper.js]: Can you please add a proper function doc

File: github_api_helper.js [hunk 4]
```diff
@@ -26,21 +26,37 @@ function inProgressAPIresponse (context, headSha) {
...
+    .filter(fileJSON => fileJSON.status !== 'removed').map(async fileInfo => {
+      return fileInfo.filename
+    })
+}
+
 /**
  * Get list of files modified by a pull request
  * @param {import('probot').Context} context Probot context
  * @param {number} number the pull request number inside the repository
  * @returns {Promise<any>} GitHub response
```
Comment (ericwb) [inline github_api_helper.js]: Doesn't look like you're returning a GitHub response object any longer.

Comment (MVrachev) [reply github_api_helper.js]: Yes, a good catch.

File: index.js [hunk 5]
```diff
@@ -102,25 +102,19 @@ async function processPullRequest (pullRequest, context) {
...
+  const filesDownloadedPromise = response
+    .map(async fileName => {
+      const headRevision = apiHelper.getContents(context, fileName, ref, pullRequest.head)
 
       // TODO: merge this code with linter-specific path resolution
-      if (filename.endsWith('.py')) {
-        cache.saveFile(repoID, id, filename, (await headRevision).data, 'python')
-      } else if (filename.endsWith('.go')) {
-        cache.saveFile(repoID, id, filename, (await headRevision).data, 'go')
+      if (fileName.endsWith('.py')) {
```
[the remark this answers is not in the document]
Comment (MVrachev) [reply index.js]: Okay, but I don't see why we can't change filename => fileName now. :D

File: test/index.test.js (hunk 1 above)
Comment (MVrachev) [reply test/index.test.js]: It used to be with big letters before but you are right.

File: config.js [hunk 6]
```diff
@@ -8,7 +8,8 @@ const config = {
   noIssuesResultTitle: 'No issues found',
   noIssuesResultSummary: 'There were no issues found.',
   issuesFoundResultTitle: 'Issues found',
-  syntaxErrorTitle: 'ERROR: Syntax error'
+  syntaxErrorTitle: 'ERROR: Syntax error',
+  numberFilesListedPerPage: 30
```
Comment (ericwb) [inline config.js]:
```suggestion
  numFilesPerPage: 30
```

Comment (MVrachev) [reply config.js]: I will change it.

**Pull request merged**